presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/csv/CsvRowEncoder.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.csv;
+
+import au.com.bytecode.opencsv.CSVWriter;
+import io.prestosql.plugin.kafka.encoder.EncoderColumnHandle;
+import io.prestosql.plugin.kafka.encoder.EncoderFieldValueProvider;
+import io.prestosql.plugin.kafka.encoder.RowEncoder;
+import io.prestosql.spi.PrestoException;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.prestosql.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.function.Function.identity;
+
+public class CsvRowEncoder
+        implements RowEncoder
+{
+    public static final String NAME = "csv";
+
+    private final Map<EncoderColumnHandle, CsvColumnEncoder> columnEncoders;
+
+    public CsvRowEncoder(Set<EncoderColumnHandle> columns)
+    {
+        columnEncoders = columns.stream().collect(toImmutableMap(identity(), this::createColumnEncoder));
+    }
+
+    private CsvColumnEncoder createColumnEncoder(EncoderColumnHandle columnHandle)
+    {
+        return new CsvColumnEncoder(columnHandle);
+    }
+
+    @Override
+    public byte[] encodeRow(List<EncoderFieldValueProvider> rowData)
+    {
+        try (StringWriter csvStr = new StringWriter();
+                CSVWriter writer = new CSVWriter(csvStr, ',', '"', "")) {
+            String[] row = new String[rowData.size()];
+            for (int i = 0; i < rowData.size(); ++i) {
+                EncoderFieldValueProvider fieldValueProvider = rowData.get(i);
+                row[i] = columnEncoders.get(fieldValueProvider.getColumn()).encodeField(fieldValueProvider);
+            }
+            writer.writeNext(row);
+            writer.flush();
+            return csvStr.toString().getBytes(UTF_8);
```

Comment:
It's more efficient to use a `ByteArrayOutputStream` as a sink and avoid creating a string.